### PR TITLE
vulkaninfo: VK_EXT_fragment_shader_interlock

### PR DESCRIPTION
**Pull request by charles-lunarg**
Adds support for the extension.

Change-Id: Ifc18acbdeae3b1b5b568e9a3ddaffb444d14ef86

### CODE REVIEW

File: vulkaninfo/vulkaninfo.c [hunk 1]
```diff
@@ -980,7 +980,9 @@ static void AppGpuInit(struct AppGpu *gpu, struct AppInstance *inst, uint32_t id
             {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DESCRIPTOR_INDEXING_PROPERTIES_EXT,
              .mem_size = sizeof(VkPhysicalDeviceDescriptorIndexingPropertiesEXT)},
             {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DEPTH_STENCIL_RESOLVE_PROPERTIES_KHR,
-             .mem_size = sizeof(VkPhysicalDeviceDepthStencilResolvePropertiesKHR)}};
+             .mem_size = sizeof(VkPhysicalDeviceDepthStencilResolvePropertiesKHR)},
+            {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_SHADER_INTERLOCK_FEATURES_EXT,
+             .mem_size = sizeof(VkPhysicalDeviceFragmentShaderInterlockFeaturesEXT)}};
```
[the remark this answers is not in the document]
Comment (charles-lunarg) [reply vulkaninfo/vulkaninfo.c]: fixed.

File: vulkaninfo/vulkaninfo.c [hunk 2]
```diff
@@ -1104,7 +1106,9 @@ static void AppGpuInit(struct AppGpu *gpu, struct AppInstance *inst, uint32_t id
             {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_YCBCR_IMAGE_ARRAYS_FEATURES_EXT,
              .mem_size = sizeof(VkPhysicalDeviceYcbcrImageArraysFeaturesEXT)},
             {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_HOST_QUERY_RESET_FEATURES_EXT,
-             .mem_size = sizeof(VkPhysicalDeviceHostQueryResetFeaturesEXT)}};
+             .mem_size = sizeof(VkPhysicalDeviceHostQueryResetFeaturesEXT)},
+            {.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_SHADER_INTERLOCK_FEATURES_EXT,
+             .mem_size = sizeof(VkPhysicalDeviceFragmentShaderInterlockFeaturesEXT)}};
```
[the remark this answers is not in the document]
Comment (charles-lunarg) [reply vulkaninfo/vulkaninfo.c]: Makes sense. Somehow didn't read the labels very well

File: vulkaninfo/vulkaninfo.c [hunk 3]
```diff
@@ -3389,6 +3393,35 @@ static void AppGpuDumpFeatures(const struct AppGpu *gpu, FILE *out) {
...
+                            "\t\t\t\t\t\t<details><summary>fragmentShaderSampleInterlock    = <span "
+                            "class='val'>%" PRIuLEAST32 "</span></summary></details>\n",
+                            fragment_shader_features->fragmentShaderSampleInterlock);
+                    fprintf(out,
+                            "\t\t\t\t\t\t<details><summary>fragmentShaderShadingRateInterlock    = <span "
+                            "class='val'>%" PRIuLEAST32 "</span></summary></details>\n",
+                            fragment_shader_features->fragmentShaderShadingRateInterlock);
+                } else if (human_readable_output) {
+                    printf("\nVkPhysicalDeviceFragmentShaderInterlockFeaturesEXT:\n");
+                    printf("=======================================\n");
```
[the remark this answers is not in the document]
Comment (charles-lunarg) [reply vulkaninfo/vulkaninfo.c]: Not all of the string's of = go all the way. Would be good to bring it back up to uniformity.

File: vulkaninfo/vulkaninfo.c [hunk 4]
```diff
@@ -3389,6 +3391,35 @@ static void AppGpuDumpFeatures(const struct AppGpu *gpu, FILE *out) {
...
+                            "\t\t\t\t\t\t<details><summary>fragmentShaderSampleInterlock    = <span "
+                            "class='val'>%" PRIuLEAST32 "</span></summary></details>\n",
+                            fragment_shader_features->fragmentShaderSampleInterlock);
+                    fprintf(out,
+                            "\t\t\t\t\t\t<details><summary>fragmentShaderShadingRateInterlock    = <span "
+                            "class='val'>%" PRIuLEAST32 "</span></summary></details>\n",
+                            fragment_shader_features->fragmentShaderShadingRateInterlock);
+                } else if (human_readable_output) {
+                    printf("\nVkPhysicalDeviceFragmentShaderInterlockFeaturesEXT:\n");
+                    printf("=====================================================\n");
```
Comment (jeremyk-lunarg) [inline vulkaninfo/vulkaninfo.c]: The underline is 2 too long since the printf command starts with `\n`.

Comment (charles-lunarg) [reply vulkaninfo/vulkaninfo.c]: Oh, that would explain all the other lines not being the same length.
I'll hope on it here and in the other PR

**Pull request merged**